src/lib/annotations/doc/DocDrawingThread.js

```diff
@@ -81,13 +90,139 @@ class DocDrawingThread extends DrawingThread {
+        }
+    }
+
+    /**
+     * Display the document drawing thread. Will set the drawing context if the scale has changed since the last show.
+     *
+     * @return {void}
+     */
+    show() {
+        if (!this.annotatedElement || !this.location) {
+            return;
+        }
+
+        // Get DrawingPath objects to be shown
+        const drawings = this.getDrawings();
+        if (this.pendingPath && !this.pendingPath.isEmpty()) {
+            drawings.push(this.pendingPath);
+        }
+
+        this.checkAndHandleScaleUpdate();
+
+        // Get the annotation layer context to draw with
+        let context;
+        if (this.state === STATES.pending) {
+            context = this.drawingContext;
+        } else {
+            this.pageEl = docAnnotatorUtil.getPageEl(this.annotatedElement, this.location.page);
+            const config = { scale: this.lastScaleFactor };
+            context = docAnnotatorUtil.getContext(
+                this.pageEl,
+                CLASS_ANNOTATION_LAYER_DRAW,
+                PAGE_PADDING_TOP,
+                PAGE_PADDING_BOTTOM
+            );
+            this.setContextStyles(config, context);
+        }
+
+        // Draw the paths to the annotation layer canvas
+        if (context) {
+            context.beginPath();
+            drawings.forEach((drawing) => {
+                drawing.generateBrowserPath(this.reconstructBrowserCoordFromLocation);
+                drawing.drawPath(context);
+            });
+            context.stroke();
+        }
+    }
+
+    /**
+     * Prepare the pending drawing canvas if the scale factor has changed since the last render. Will do nothing if
+     * the thread has not been assigned a page.
+     *
+     * @return {void}
+     */
+    checkAndHandleScaleUpdate() {
+        const scale = annotatorUtil.getScale(this.annotatedElement);
+        if (this.lastScaleFactor === scale || (!this.location || !this.location.page)) {
+            return;
+        }
+
+        // Set the scale and in-memory context for the pending thread
+        this.lastScaleFactor = scale;
+        this.pageEl = docAnnotatorUtil.getPageEl(this.annotatedElement, this.location.page);
+        this.drawingContext = docAnnotatorUtil.getContext(
+            this.pageEl,
+            CLASS_ANNOTATION_LAYER_DRAW_IN_PROGRESS,
+            PAGE_PADDING_TOP,
+            PAGE_PADDING_BOTTOM
+        );
+
+        const config = { scale };
+        this.setContextStyles(config);
+    }
+
+    /**
+     * Requires a DocDrawingThread to have been started with DocDrawingThread.start(). Reconstructs a browserCoordinate
+     * relative to the dimensions of the DocDrawingThread page element.
+     *
+     * @private
+     * @param {Location} documentLocation - The location coordinate relative to the document
+     * @return {Location} The location coordinate relative to the browser
+     */
+    reconstructBrowserCoordFromLocation(documentLocation) {
```

Comment:
I think this is not generic. This is a really specific case where we need to create a location from the thread information and use docAnnotatorUtil.getBrowserCoordinatesFromLocation(). Its sort of a shallow wrapper for creating a location, passing a location into the docannotatorutil function and then creating a new location based on the return of docAnnotatorUtil.getBrowserCoord.